src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +33,115 @@ public PropertySubpatternCompletionProvider()
-                // PROTOTYPE(extended-property-patterns) ExpressionColon
-                p => p.NameColon?.Name.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
+            if (propertyPatternClause is not null)
+            {
+                // Filter out those members that have already been typed as simple (not extended) properties
+                var alreadyTestedMembers = new HashSet<string>(propertyPatternClause.Subpatterns.Select(
+                    p => p.NameColon?.Name.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
 
-            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name) &&
-                m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+                members = members.Where(m => !alreadyTestedMembers.Contains(m.Name));
+            }
 
-            foreach (var untestedMember in untestedMembers)
+            foreach (var member in members)
             {
                 const string ColonString = ":";
                 context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
-                    displayText: untestedMember.Name.EscapeIdentifier(),
+                    displayText: member.Name.EscapeIdentifier(),
                     displayTextSuffix: ColonString,
                     insertionText: null,
-                    symbols: ImmutableArray.Create(untestedMember),
-                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    symbols: ImmutableArray.Create(member),
+                    contextPosition: context.Position,
                     rules: s_rules));
             }
+
+            return;
+
+            static ITypeSymbol GetMemberAccessType(ITypeSymbol type, ExpressionSyntax expression, Document document, SemanticModel semanticModel, int position)
+            {
+                string name;
+                if (expression is MemberAccessExpressionSyntax memberAccess)
+                {
+                    type = GetMemberAccessType(type, memberAccess.Expression, document, semanticModel, position);
+                    name = memberAccess.Name.Identifier.ValueText;
+                }
+                else if (expression is IdentifierNameSyntax identifier)
+                {
+                    name = identifier.Identifier.ValueText;
+                }
+                else
+                {
+                    return null;
+                }
+
+                return GetMemberType(type, name, document, semanticModel, position);
+            }
+
+            static ITypeSymbol GetMemberType(ITypeSymbol type, string name, Document document, SemanticModel semanticModel, int position)
+            {
+                var members = GetCandidatePropertiesAndFields(document, position, semanticModel, type);
+                var matches = members.Where(m => m.Name == name).ToArray();
```

Comment:
```suggestion
                var matches = members.WhereAsArray(m => m.Name == name);
```

Reply:
why not pass in 'name' into GetCandidatePropertiesAndFields?

Reply:
> why not pass in 'name' into GetCandidatePropertiesAndFields?

This local function is used twice. One of the usages doesn't have a name to filter on.

Reply:
> WhereAsArray

I don't think there's such extension on IEnumerable

Reply:
this can use ImmutableArray from bottom to top :)